pkg/rbac/rbac.go

```
@@ -0,0 +1,287 @@
+
+	return resourceAccessResults, nil
+}
+
+// a convenience method for building a list with unique entries
+// specified items are added to the given  slice  of items if not already in it
+
+func addUniqueItems(itemlist []string, itemsToAdd ...string) []string {
+	klog.V(2).Infof("Adding items %v to list %v", itemsToAdd, itemlist)
+
+	// iterate through each of the items in  itemsToAdd list
+	// and add item only  if it doesnt already exist in the itemlist
+	resultList := itemlist
+	for _, item := range itemsToAdd {
+		// check if itemList  contains the item before appending
+		if !slices.Contains(resultList, item) {
+			resultList = append(resultList, item)
+		}
+	}
+
+	klog.V(2).Infof("Result List: %v ", resultList)
+
+	return resultList
+}
+
+// makes a selfsubjectrulesreview call , the call requires a namespace to be set
+// call returns the list of the allowed acls for the user
+// for namespace-scoped resources in a given namespace +  acls for  all cluster-scoped resources
+// if no namespace is set, e.g. if the user only cares about the acls for cluster scoped resources,
+// then the namespace defaults to  "default"
+// When the user's bearer token is used to make the call instead of  impersonation,
+// the user's usergroup are already taken into account when returning the list of acls
+// if the call is successful then this function returns a list of resourcerules
+func MakeSubjectAccessRulesReviewForUser(
+	kclient kubernetes.Interface, namespace string,
+) ([]authorizationv1.ResourceRule, error) {
+	klog.V(2).Infof("Make Subject Access Rules Review for Namespace %s", namespace)
+
+	if len(namespace) == 0 {
+		namespace = "default"
+	}
+	// make api call
+	sarr := &authorizationv1.SelfSubjectRulesReview{
+		Spec: authorizationv1.SelfSubjectRulesReviewSpec{
+			Namespace: namespace,
+		},
+	}
+
+	response, err := kclient.AuthorizationV1().SelfSubjectRulesReviews().Create(
+		context.TODO(), sarr, metav1.CreateOptions{})
+	if err != nil {
+		log.Fatal(err)
```

Shouldn't we just return the error here?

---

yes, should return err, fixed it thanks